src/ipc/handler.c

```diff
@@ -720,16 +721,21 @@ static int ipc_dma_trace_config(uint32_t header)
 	uint32_t ring_size;
 #endif
 	struct sof_ipc_dma_trace_params_ext params;
+	uint64_t ticks;
 	int err;
 
 	/* copy message with ABI safe method */
 	IPC_COPY_CMD(params, _ipc->comp_data);
 
-	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT)
-		platform_timer->delta = params.timestamp -
+	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT) {
+		ticks = clock_ns_to_ticks(platform_generic_queue
```

Comment:
This here is too complex, you can reference `clock_ms_to_ticks` user to simplify here.

Reply:
@xiulipan Updated. But I don't know if using ``PLATFORM_DEFAULT_CLOCK`` is always right. for example, On baytrail, the .clk is [PLATFORM_WORKQ_CLOCK](https://github.com/thesofproject/sof/blob/master/src/platform/baytrail/platform.c#L145), but the PLATFORM_DEFAULT_CLOCK is  [CLK_SSP](https://github.com/thesofproject/sof/blob/master/src/platform/baytrail/include/platform/platform.h#L62)

Reply:
@bardliao 
Then maybe you can take reference of `sch->clock` to pre define it for different platform.

Reply:
@xiulipan updated.

Reply:
Existing `clock_ms_to_ticks()` is used with the `PLATFORM_DEFAULT_CLOCK` in many places assuming this is correct define.
Wrt to BYT see this define:
```c
/* platform WorkQ clock */
#define PLATFORM_WORKQ_CLOCK	PLATFORM_DEFAULT_CLOCK
```